docs/source/models/segment_anything.rst

```diff
@@ -0,0 +1,255 @@
+Segment Anything (SAM)
+======================
+
+The Segment Anything Model (SAM) produces high quality object masks from input prompts such as points or boxes, and it
+can be used to generate masks for all objects in an image.
+
+.. card::
+    :link: https://segment-anything.com/
+
+    **Segment Anything**
+    ^^^
+    **Abstract:** We introduce the Segment Anything (SA) project: a new task, model, and dataset for image
+    segmentation. Using our efficient model in a data collection loop, we built the largest segmentation
+    dataset to date (by far), with over 1 billion masks on 11M licensed and privacy respecting images. The
+    model is designed and trained to be promptable, so it can transfer zero-shot to new image distributions
+    and tasks. We evaluate its capabilities on numerous tasks and find that its zero-shot performance is impressive
+    -- often competitive with or even superior to prior fully supervised results. We are releasing the Segment Anything
+    Model (SAM) and corresponding dataset (SA-1B) of 1B masks and 11M images at https://segment-anything.com to foster
+    research into foundation models for computer vision.
+
+    **Tasks:** Segmentation
+
+    **Datasets:** SA-1B
+
+    **Licence:** Apache
+
+    +++
+    **Authors:** Alexander Kirillov and Eric Mintun and Nikhila Ravi and Hanzi Mao and Chloe Rolland and Laura
+    Gustafson and Alex Berg and Wan-Yen Lo and Piotr Dollar and Ross Girshick
+
+
+
+How to use SAM from Kornia
+--------------------------
+
+Prompter
+^^^^^^^^
+.. _anchor Prompter:
+
+The High level API `ImagePrompter` handle with the image and prompt transformation, preprocessing and prediction for
+a given SAM model.
+
+.. code-block:: python
+
+    import torch
+
+    from kornia.contrib import SamConfig
+    from kornia.contrib.prompter import ImagePrompter
+    from kornia.io import load_image, ImageLoadType
+    from kornia.geometry.keypoints import Keypoints
+    from kornia.geometry.boxes import Boxes
+    from kornia.utils import get_cuda_device_if_available
+
+    model_type = 'vit_h' # or can be a number `0` or the enum kornia.contrib.sam.SamModelType.vit_h
+    checkpoint_path = './path_for_the_vit_h_checkpoint.pth'
```

Comment:
can we show in the first place that the model will get downloaded automatically ?